earthaccess/results.py

```diff
@@ -244,6 +244,9 @@ def _repr_html_(self) -> str:
         granule_html_repr = _repr_granule_html(self)
         return granule_html_repr
 
+    def __hash__(self) -> int:
+        return hash(self["meta"]["concept-id"])
```

Comment:
@betolink @chuckwondo This seems reasonable, but please validate me :)

Reply:
Thinking about it for like 5 minutes, this is obviously a bad idea. This class is subclassing `dict`. We'd need to implement like a `frozendict`.